azurerm plugin crashes on adding azurerm_application_insights_api_key

<!---
Please note the following potential times when an issue might be in Terraform core:

* [Configuration Language](https://www.terraform.io/docs/configuration/index.html) or resource ordering issues
* [State](https://www.terraform.io/docs/state/index.html) and [State Backend](https://www.terraform.io/docs/backends/index.html) issues
* [Provisioner](https://www.terraform.io/docs/provisioners/index.html) issues
* [Registry](https://registry.terraform.io/) issues
* Spans resources across multiple providers

If you are running into one of these scenarios, we recommend opening an issue in the [Terraform core repository](https://github.com/hashicorp/terraform/) instead.
--->

<!--- Please keep this note for the community --->

### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

<!--- Thank you for keeping this note for the community --->

### Terraform (and AzureRM Provider) Version

<!--- Please run `terraform -v` to show the Terraform core version and provider version(s). If you are not running the latest version of Terraform or the provider, please upgrade because your issue may have already been fixed. [Terraform documentation on provider versioning](https://www.terraform.io/docs/configuration/providers.html#provider-versions). --->

### Affected Resource(s)
* `Terraform v1.1.2 on windows_amd64`
* `azurerm_2.94.0`
* `azurerm_2.95.0`
* `azurerm_application_insights_api_key`
<!--- Please list the affected resources and data sources. --->

### Terraform Configuration Files

<!--- Information about code formatting: https://help.github.com/articles/basic-writing-and-formatting-syntax/#quoting-code --->

```hcl
terraform {

  required_providers {
    azurerm = {
      source  = "hashicorp/azurerm"
      version = "2.95.0"
    }
  }
  required_version = ">= 1.0.0"
}

provider "azurerm" {
    subscription_id = "xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx"
  features {}
}

resource "azurerm_resource_group" "rg" {
  name     = "rg-appinsights-bug-test2"
  location = "West Europe"
}

resource "azurerm_application_insights" "applicationinsights" {
  name                = "test-app-inisghts"
    resource_group_name = azurerm_resource_group.rg.name
  location            = azurerm_resource_group.rg.location

  application_type    = "web"
}
resource "azurerm_application_insights_api_key" "read_telemetry" {
  name                    = "read-telemetry-and-annotations"
  application_insights_id = azurerm_application_insights.applicationinsights.id
   read_permissions        = ["aggregate", "api", "draft", "extendqueries", "search"]
  write_permissions       = ["annotations"]
}
```

### Panic Output
https://gist.github.com/bamcore/bea13ec3a470393d51a870c34b58f0b9

### Expected Behaviour
The azurerm provider should have created the application inisghts api key "read-telemetry-and-annotations"
<!--- What should have happened? --->

### Actual Behaviour
panic: runtime error: invalid memory address or nil pointer dereference
[signal 0xc0000005 code=0x0 addr=0x0 pc=0x4cfa76d]
<!--- What actually happened? --->

### Steps to Reproduce

<!--- Please list the steps required to reproduce the issue. --->

1. `terraform apply`

--- a/internal/services/applicationinsights/application_insights_api_key_resource.go
+++ b/internal/services/applicationinsights/application_insights_api_key_resource.go
@@ -102,6 +102,7 @@ func resourceApplicationInsightsAPIKeyCreate(d *pluginsdk.ResourceData, meta int
 	name := d.Get("name").(string)
 
 	var existingAPIKeyList insights.ApplicationInsightsComponentAPIKeyListResult
+	var existingAPIKeyId *parse.ApiKeyId
 	var keyId string
 	existingAPIKeyList, err = client.List(ctx, appInsightsId.ResourceGroup, appInsightsId.Name)
 	if err != nil {
@@ -111,7 +112,7 @@ func resourceApplicationInsightsAPIKeyCreate(d *pluginsdk.ResourceData, meta int
 	}
 
 	for _, existingAPIKey := range *existingAPIKeyList.Value {
-		existingAPIKeyId, err := parse.ApiKeyID(camelCaseApiKeys(*existingAPIKey.ID))
+		existingAPIKeyId, err = parse.ApiKeyID(camelCaseApiKeys(*existingAPIKey.ID))
 		if err != nil {
 			return err
 		}
@@ -131,8 +132,8 @@ func resourceApplicationInsightsAPIKeyCreate(d *pluginsdk.ResourceData, meta int
 		}
 	}
 
-	if !utils.ResponseWasNotFound(existing.Response) {
-		return tf.ImportAsExistsError("azurerm_application_insights_api_key", *existing.ID)
+	if !utils.ResponseWasNotFound(existing.Response) && existingAPIKeyId != nil {
+		return tf.ImportAsExistsError("azurerm_application_insights_api_key", existingAPIKeyId.ID())
 	}
 
 	apiKeyProperties := insights.APIKeyRequest{

